### PR TITLE
chore: convert `tasks` directory to `vitest`

### PR DESCRIPTION
**Pull request by AtofStryker**
- Closes 

### Additional details

Converts the tasks directory to vitest. Some of the source files in the directory have also been reworked to make them easier to test.

A few notable things here. The ListR default renderer does not work in service worker processes and falls back to the verbose renderer, which actually works better for us as we are able to determine our snapshots in a more deterministic way

I also needed to change the way `verify` exports its functions as it was causing issues when running the CLI after compilation.

I also needed to change the default scripts for testing mocha as you can't run an empty mocha test suite. I will be cleaning this up in a follow up PR

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/test/lib/tasks/__snapshots__/install.spec.ts.snap [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+https://on.cypress.io/app/get-started/install-cypress#System-requirements
+
+----------
+
+Platform: win32-ia32
+"
+`;
+
+exports[`/lib/tasks/install > .start > is silent when log level is silent > silent install 1 1`] = `
+"[STARTED] Task without title.
```
Comment (AtofStryker) [inline cli/test/lib/tasks/__snapshots__/install.spec.ts.snap:15]: I left a detailed comment in the spec, but the `Listr` renderer that is used in Vitest is different than the renderer used in Mocha. This is due to TTY not being supported inside of service worker processes, so Vitest defaults back to the VerboseRenderer vs the DefaultRenderer. The VerboseRenderer is likely better for us in a snapshot context because the text printed is deterministic as opposed to icons being constantly replaced in the terminal 

see https://listr2.kilic.dev/renderer/renderer.html#frontmatter-title

**Review by cacieprins — APPROVED** [review]
